sdk/provisioning/Azure.Provisioning/src/Primitives/Resource.cs

```diff
@@ -35,6 +36,29 @@ public abstract class Resource(string resourceName, ResourceType resourceType, s
     /// </summary>
     public bool IsExistingResource { get; protected set; } = false;
 
+    /// <summary>
+    /// Declares explicit dependencies on other resources.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// While you may be inclined to use dependsOn to map relationships between
+    /// your resources, it's important to understand why you're doing it.  For
+    /// example, to document how resources are interconnected, dependsOn isn't
+    /// the right approach. After deployment, the resource doesn't retain
+    /// deployment dependencies in its properties, so there are no commands or
+    /// operations that let you see dependencies. Setting unnecessary
+    /// dependencies slows deployment time because Resource Manager can't
+    /// deploy those resources in parallel.
+    /// </para>
+    /// <para>
+    /// Even though explicit dependencies are sometimes required, the need for
+    /// them is rare. In most cases, you can use a symbolic name to imply the
+    /// dependency between resources. If you find yourself setting explicit
+    /// dependencies, you should consider if there's a way to remove it.
+    /// </para>
+    /// </remarks>
+    public IList<Resource> DependsOn { get; } = [];
```

Comment:
Is `Resource` good enough? Will this ever need to be an Expression or similar?

Reply:
It can only be resource names.  I'll make a note to circle back to this before GA to see if we want to support `ResourceIdentifier` somehow as well, but given it's a niche scenario I'd lean toward making people use `Foo.FromExisting("...")` if they wanted to add a dependency to an external resource.